chore(renovate): pin monaco-editor `expressions` and `entities-routes`

sumimakito opened the pull request:
# Summary

This PR tries to pin `monaco-editor` by excluding it for `@kong-ui-public/expressions` and `@kong-ui-public/entities-routes` via `packageRules` in `renovate.json`.

### Why?

`monaco-editor` is currently only used for the Expressions routes support in `@kong-ui-public/expressions` and `@kong-ui-public/entities-routes`. We encountered several issues while consuming the latest `monaco-editor` in the host app during the development. To deliver the Expressions routes support in time, we decided to downgrade it to [`v0.21.3`–which we used previously in Kong Manager](https://github.com/Kong/kong-admin/blob/9bae765787d9ea28cfa48e16bd3c56b65e82a2c7/package.json#L85) to prevent [the Renovate bot from bumping it](https://github.com/Kong/public-ui-components/pull/1352). 

We will try to update it to the latest possible minor release after Kong Gateway 3.7's release.

adamdehaven commented on renovate.json:
```diff
@@ -7,5 +7,6 @@
     "pnpm": "8.6.11"
-  }
+  },
+  "ignoreDeps": ["monaco-editor"]
```
Yes, sounds good to me

mptap reviewed (commented):
Could we expedite this PR? 
Seems its needed to unblock upgrading `@kong-ui-public/entities-routes` 
See example renovate pr failure https://github.com/Kong/konnect-ui-apps/actions/runs/8758729274

sumimakito commented:
@mptap Sure. I've updated this PR to use `packageRules` to scope the rule to `@kong-ui-public/expressions` and `@kong-ui-public/entities-routes` only. 

Meanwhile, I don't feel the CI failure in the recent Renovate bumping PR (https://github.com/Kong/konnect-ui-apps/pull/3018) was caused by this change. Instead, it was because the Expressions editor component in `@kong-ui-public/entities-routes` imports `monaco-editor`, but we made `monaco-editor` a peer dependency instead of bundling it. To fix this, I've pushed a commit (https://github.com/Kong/konnect-ui-apps/pull/3018/commits/c65e71509076678a6566649ccb55a7662706f55a) to the Renovate PR to add `monaco-editor` as external in the Rollup config, as it is optional for us as of now (we don't show/use the Expressions editor at the moment), and I also updated several type imports from `@kong-ui-public/entities-routes`.

Leopoldthecoder reviewed (approved):
LGTM

kaiarrowood reviewed (dismissed):
LGTM! 🚀

adamdehaven commented on renovate.json:
```diff
@@ -7,5 +7,12 @@
     "pnpm": "8.6.11"
-  }
+  },
+  "packageRules": [
+    {
+      "matchPackageNames": ["@kong-ui-public/expressions", "@kong-ui-public/entities-routes"],
+      "groupName": "exclude monaco-editor from @kong-ui-public/expressions and @kong-ui-public/entities-routes",
+      "excludeDepNames": ["monaco-editor"]
+    }
+  ]
```
issue: `matchPackageNames` is for the matched dependency package names, not the packages within the repository

I think what you actually want here is something like `matchFileNames` whereby you provide the path to the `package.json` of these packages, e.g. 

```json
{
  "packageRules": [
    {
      "description": "Pin monaco-editor dependency for the expressions package",
      "matchFileNames": ["packages/core/expressions/package.json"],
      "groupName": "expressions-monaco-editor",
      "matchPackageNames": ["monaco-editor"],
      "enabled": false
    }
  ]
}
```

sumimakito replied:
@adamdehaven Thanks. I've updated the configuration with the suggestion
